repository-module/src/main/java/com/gmail/ezlotnikova/repository/ShopRepository.java

```diff
@@ -0,0 +1,11 @@
+package com.gmail.ezlotnikova.repository;
+
+import java.util.List;
+
+import com.gmail.ezlotnikova.repository.model.Shop;
+
+public interface ShopRepository extends GenericRepository<Long, Shop> {
+
+    List<Shop> findShopsWithItem(Long id);
```

Comment:
No need to use custom method

Reply:
Homework was re-writed wtih new unidirectional many-to-many relationship (from item to shop). All unnesesary methods and implementations deleted.